src/ngraph/op/gather.cpp

```diff
@@ -82,3 +95,119 @@ void op::Gather::validate_and_infer_types()
 
     set_output_type(0, result_et, result_shape);
 }
+
+void op::v0::Gather::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVector& deltas)
+{
+    throw ngraph_error("Not yet implemented");
+}
+
+const string op::v1::Gather::type_name{"Gather"};
+
+op::v1::Gather::Gather(const Output<Node>& params,
+                       const Output<Node>& indices,
+                       const Output<Node>& axes)
+    : Op({params, indices, axes})
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v1::Gather::validate_and_infer_types()
+{
+    const auto& input_rank = get_input_partial_shape(PARAMS).rank();
+    const auto& axis_shape = get_input_partial_shape(AXIS);
+    const auto& axis_rank = axis_shape.rank();
+
+    if (axis_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(axis_rank) == 1 &&
+                                  static_cast<size_t>(axis_shape[0]) == 1,
```

Comment:
I think this line will not work if `axis_shape.is_static() == false`. You should add this check to the `if` above.